react/features/connection-indicator/components/native/ConnectionIndicator.js

```diff
@@ -0,0 +1,63 @@
+// @flow
+
+import React from 'react';
+
+import { BaseIndicator } from '../../../base/react';
+import { connect } from '../../../base/redux';
+
+import AbstractConnectionIndicator, {
+    type Props,
+    type State
+} from '../AbstractConnectionIndicator';
+
+import { CONNECTOR_INDICATOR_COLORS } from './styles';
+
+/**
+ * Implements ain indicator to show the quality of the connection of a participant.
+ */
+class ConnectionIndicator extends AbstractConnectionIndicator<Props, State> {
+    /**
+     * Initializes a new {@code ConnectionIndicator} instance.
+     *
+     * @param {Object} props - The read-only properties with which the new
+     * instance is to be initialized.
+     */
+    constructor(props: Props) {
+        super(props);
+
+        this.state = {
+            autoHideTimeout: undefined,
+            showIndicator: false,
+            stats: {}
+        };
+    }
+
+    /**
+     * Implements React's {@link Component#render()}.
+     *
+     * @inheritdoc
+     * @returns {ReactElement}
+     */
+    render() {
+        const { showIndicator, stats } = this.state;
+        const { percent } = stats;
+
+        if (!showIndicator || typeof percent === 'undefined') {
+            return null;
+        }
+
+        // Signal level on a scale 0..2
+        const signalLevel = Math.floor(percent / 33.4);
+
+        return (
+            <BaseIndicator
+                icon = { `signal_cellular_${signalLevel}` }
+                iconStyle = {{
+                    color: CONNECTOR_INDICATOR_COLORS[signalLevel]
```

Comment:
Will you be updating web to use the indicator?  What about the ninja icon?

Reply:
I didn't plan to change web in this regard. The reason why I used a new icon is that web used some css magic to render the indicator which I didn't want to replicate on mobile (probably would't even be able to). Do you think I should change web too? If so, then I need a confirmation from Yana, because adding a new component and styling it on my own is one thing, but redesigning an existing one is another. What do you think?

Regarding the ninja icon (and cc-in @saghul here as he also asked about it): I really don't want to have it on mobile for various reasons:
1. What the f* is a ninja icon? I get the functionality, but really? A ninja?
2. The Thumbnail already looks like a Christmas tree even when only half of the indicators are on. This is not a good example of a "clean app" for me.

Reply:
If we are adding the same behavior as web, then it should behave the same. If we were building a different UX, that'd be another story IMHO.

Also, regarding the ninja, this is one case where I think consistency trumps correctness. Seeing the same thing on web and mobile is A Good Thing (TM).

Reply:
If swapping the connection strength icon has not already been planned then I think we should ask and plan it for a future change if we get confirmation it is desired. It’s a small consistency win if web and mobile share the same indicator, and maybe simpler code due to a general lack of css familiarity.